init/eessi_archdetect.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# current pathways implemented in EESSI
+# x86_64/generic
+# x86_64/intel/haswell
+# x86_64/intel/skylake_avx512
+# x86_64/amd/zen2
+# x86_64/amd/zen3
+# aarch64/generic
+# ppc64le/generic
+# ppc64le/power9le
+
+ARGUMENT=${1:-none}
+
+cpupath () {
+  # let the kernel tell base machine type
+  MACHINE_TYPE=$(uname -m)
+
+  # fallback path
+  CPU_PATH="${MACHINE_TYPE}/generic"
+
+  if [ ${MACHINE_TYPE} == "aarch64" ]; then
+    echo ${CPU_PATH}
+    exit
+  fi
+
+  if [ ${MACHINE_TYPE} == "ppc64le" ]; then
+    echo ${CPU_PATH} "not sure what to do next..."
+    echo "please mail output of lscpu to me..."
+    exit
+  fi
+
+  if [ ${MACHINE_TYPE} == "x86_64" ]; then
+    # check for vendor info, if available, for x86_64
+    CPUINFO_VENDOR_FLAG=$(grep -m 1 ^vendor_id /proc/cpuinfo)
+    [[ $CPUINFO_VENDOR_FLAG =~ .*GenuineIntel* ]] && CPU_VENDOR=intel
+    [[ $CPU_VENDOR_FLAG =~ .*AuthenticAMD* ]] && CPU_VENDOR=amd
+
+    CPU_FLAGS=$(grep -m 1 ^flags /proc/cpuinfo)
+    [[ $CPU_FLAGS =~ .*avx2* ]] && HAS_AVX2=true
```

Comment:
We need to protect ourselves from not picking up `$HAS_AVX2` that is set outside of this script, by declaring these local?

Reply:
done